src/Bicep.Core/Emit/ParametersFileTemplateWriter.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Semantics;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+
+namespace Bicep.Core.Emit
+{
+    public class ParametersFileTemplateWriter : TemplateWriter
+    {
+        public ParametersFileTemplateWriter(SemanticModel semanticModel, EmitterSettings settings): base(semanticModel, settings)
+        {
+            this.context = new EmitterContext(semanticModel, settings);
+            this.settings = settings;
+        }
+
+        private readonly EmitterContext context;
+        private readonly EmitterSettings settings;
+
+        public new void Write(JsonTextWriter writer)
+        {
+            // Template is used for calcualting template hash, template jtoken is used for writing to file.
+            var templateJToken = GenerateTemplate();
+            if (templateJToken.SelectToken(GeneratorMetadataPath) is not JObject generatorObject)
+            {
+                throw new InvalidOperationException($"generated template doesn't contain a generator object at the path {GeneratorMetadataPath}");
+            }
+            templateJToken.WriteTo(writer);
+        }
+
+        private JToken GenerateTemplate()
+        {
+            // TODO: since we merely return a JToken, refactor the emitter logic to add properties to a JObject
+            // instead of writing to a JsonWriter and converting it to JToken at the end
+            using var stringWriter = new StringWriter();
+            using var jsonWriter = new JsonTextWriter(stringWriter);
+            var emitter = new ExpressionEmitter(jsonWriter, this.context);
+
+            jsonWriter.WriteStartObject();
+
+            emitter.EmitProperty("$schema", "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#");
+
+            emitter.EmitProperty("contentVersion", "1.0.0.0");
+
+            this.EmitMetadata(jsonWriter, emitter);
+
+            this.EmitParametersIfPresent(jsonWriter, emitter);
```

Comment:
>EmitParametersIfPresent [](http://example.com/codeflow?start=17&length=23)

Are you sure this is the right method to call here? This is the method from the template writer that writes out the parameter schema including `type` properties which don't belong in the parameters file, which should only have parameter values.

Reply:
+1

Reply:
you're right, and according to the official 'parameters file' documentation, the json file shouldn't include 'type' fields.

I sent an update to the [ParametersFileTemplateWriter.cs](https://github.com/Azure/bicep/pull/6601/files/c4ef9ec5347315890d6c501f7adb5c793fcd5c0c#diff-30a900071270f3f38861de1729f2ccb8d1776b6467b266f561c7b255a50eaba3) and changed the way to emit parameter fields.

That change should resolve your comment, thanks 👍